### DISCUSSION TITLE
Dark mode for PymPress as shown on main page

### DISCUSSION THREAD

**Discussion by MartinRJDagleish** [Cimbali/pympress#218] [Q&A] [answered]
Hello there. 
Thanks for creating this wonderful programme! 

I wanted to ask if there is a way to get the dark mode which is shown on the main page for Windows 10. 

Currently, can't find an option or settings to set the color.

**Comment by Cimbali** [comment, marked as answer]
Ah, it’s actually the “Breeze dark” gtk theme. What you currently have is the adwaita theme, which is by default. Switching that on linux is rather easy, here’s the way to do it on windows:

1. Install a theme

    There are 2 locations, either install the theme for all your gtk apps, e.g. in `C:\Users\%USERNAME%\AppData\Local\themes`, or just for pympress, so in `%INSTALLDIR%\share\themes` (for me that’s `C:\Users\%USERNAME%\AppData\Local\Programs\pympress\share\themes`)

    Basically pick a theme [e.g. from this list](https://www.gnome-look.org/browse/cat/135/ord/rating/?tag=dark) and make sure to unpack it in the selected directory, it needs at least `%THEMENAME%\gtk-3.0\gtk.css` and `%THEMENAME%\index.theme`, where `THEMENAME` is the name of the theme.

    I’ve seen 2 pitfalls:
    - themes that are not self-contained (relying on re-using css from default linux themes that you might not have), and
    - linux links (files under gtk-3.0/ that point to a directory above and that need to be replaced by a directory containing the contents of the target directory that has the same name as the link file).

2. Set the theme as default

    Create a `settings.ini` file, either under `C:\Users\%USERNAME%\AppData\Local\gtk-3.0` (global setting) or `%INSTALLDIR%\etc\gtk-3.0` (just pympress) and set the contents:

    ```ini
    [Settings]
    gtk-theme-name=THEMENAME
    ```

Here’s what it looks like with the [Obscure-Orange](https://www.gnome-look.org/p/1254680/) theme.
![VirtualBox_Win64_16_05_2021_01_23_19](https://user-images.githubusercontent.com/6126377/118380851-70d3d080-b5e5-11eb-97ac-65961f343a2d.png)

In testing this  found these 2 stackoverflow questions useful:
- [Change GTK+3 look on Windows](https://stackoverflow.com/a/39041558/1387346) which contains a list of all interesting directories
- [How to get native windows decorations on GTK3 on Windows 7+ and MSYS2](https://stackoverflow.com/a/37060369/1387346) which details the process

**Comment by MartinRJDagleish** [comment]
Hey @Cimbali! Thanks a lot! Will try your answer and reply when I've got the time to test it, but I really appreciate you taking the time to explain and provide links! :)

**Comment by Cimbali** [comment]
I’ve made this answer [part of the docs](https://cimbali.github.io/pympress/options.html#themes-on-windows) for future reference. I think we can close this issue now (unless it doesn’t work − in which case feel free to reopen).
